wallet/src/types.rs

```diff
@@ -275,3 +297,437 @@ pub struct SuperBlockNotification {
+            collateral: x.collateral,
+        }
+    }
+}
+
+#[derive(Debug, Eq, PartialEq, Clone, Serialize, Deserialize)]
+pub struct VTTransactionHelper {
+    #[serde(
+        serialize_with = "into_generic_type::<_, VTTransactionBodyHelper, _>",
+        deserialize_with = "from_generic_type::<_, VTTransactionBodyHelper, _>"
+    )]
+    pub body: VTTransactionBody,
+    pub signatures: Vec<KeyedSignature>,
+}
+
+impl From<VTTransaction> for VTTransactionHelper {
+    fn from(x: VTTransaction) -> Self {
+        VTTransactionHelper {
+            body: x.body,
+            signatures: x.signatures,
+        }
+    }
+}
+
+impl From<VTTransactionHelper> for VTTransaction {
+    fn from(x: VTTransactionHelper) -> Self {
+        VTTransaction {
+            body: x.body,
+            signatures: x.signatures,
+        }
+    }
+}
+
+#[derive(Debug, Eq, PartialEq, Clone, Serialize, Deserialize)]
+pub struct VTTransactionBodyHelper {
+    pub inputs: Vec<TransactionInput>,
+    #[serde(
+        serialize_with = "into_generic_type_vec::<_, ValueTransferOutputHelper, _>",
+        deserialize_with = "from_generic_type_vec::<_, ValueTransferOutputHelper, _>"
+    )]
+    pub outputs: Vec<ValueTransferOutput>,
+}
+
+impl From<VTTransactionBody> for VTTransactionBodyHelper {
+    fn from(x: VTTransactionBody) -> Self {
+        VTTransactionBodyHelper {
+            inputs: x.inputs,
+            outputs: x.outputs,
+        }
+    }
+}
+
+impl From<VTTransactionBodyHelper> for VTTransactionBody {
+    fn from(x: VTTransactionBodyHelper) -> Self {
+        VTTransactionBody::new(x.inputs, x.outputs)
+    }
+}
+
+/// Value transfer output transaction data structure
+#[derive(Debug, Eq, PartialEq, Clone, Serialize, Deserialize, Hash)]
+pub struct ValueTransferOutputHelper {
+    pub pkh: PublicKeyHash,
+    #[serde(
+        serialize_with = "u64_to_string",
+        deserialize_with = "number_from_string"
+    )]
+    pub value: u64,
+    /// The value attached to a time-locked output cannot be spent before the specified
+    /// timestamp. That is, they cannot be used as an input in any transaction of a
+    /// subsequent block proposed for an epoch whose opening timestamp predates the time lock.
+    pub time_lock: u64,
+}
+
+impl From<ValueTransferOutput> for ValueTransferOutputHelper {
+    fn from(x: ValueTransferOutput) -> Self {
+        ValueTransferOutputHelper {
+            pkh: x.pkh,
+            value: x.value,
+            time_lock: x.time_lock,
+        }
+    }
+}
+
+impl From<ValueTransferOutputHelper> for ValueTransferOutput {
+    fn from(x: ValueTransferOutputHelper) -> Self {
+        ValueTransferOutput {
+            pkh: x.pkh,
+            value: x.value,
+            time_lock: x.time_lock,
+        }
+    }
+}
+
+/// Value transfer output transaction data structure
+#[derive(Debug, Eq, PartialEq, Clone, Serialize, Deserialize, Hash, Default)]
+pub struct VttOutputParamsHelper {
+    pub address: String,
```

Comment:
I think this should use the `PublicKeyHash` type, this way invalid addresses are found at deserialization time:

```suggestion
    pub address: PublicKeyHash,
```